### ISSUE TITLE
[Training] Scheduler

### ISSUE THREAD

**Issue by vladimirjovanovicTT** [tenstorrent/tt-forge-fe#994]
We need to add support for learning rate scheduling when training models, similar to this:
https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.LRScheduler.html#torch.optim.lr_scheduler.LRScheduler

For a first version, we can have a scheduler concept that links to our optimizer and contains logic for scheduling/adapting learning rate.  Optimizers need to have learning rates accessible/modifiable.
